Result/Result.swift

```diff
@@ -169,6 +185,33 @@ extension NSError: ErrorProtocolConvertible {
 /// contains an `Int`eger and is guaranteed never to be a `failure`.
 public enum NoError: Swift.Error { }
 
+/// A type-erased error which wraps an arbitrary error instance. This should be
+/// useful for generic contexts.
+public struct AnyError: Swift.Error, ErrorProtocolConvertible, Equatable, CustomStringConvertible {
+	/// The underlying error.
+	public let error: Swift.Error
+
+	public init(_ error: Swift.Error) {
+		self.error = error
+	}
+
+	public static func error(from error: Error) -> AnyError {
+		if let anyError = error as? AnyError {
+			return anyError
+		}
+		return AnyError(error)
+	}
+
+	public static func ==(lhs: AnyError, rhs: AnyError) -> Bool {
+		return lhs.error._code == rhs.error._code
+			&& lhs.error._domain == rhs.error._domain
```

Comment:
> Are `_domain` and `_code` allowed to use?

I'm not confident. It should be okay to remove `Equatable` conformance (in that case use them for testing purpose only).